core/client/fs/src/main/java/alluxio/client/file/BaseFileSystem.java

```diff
@@ -288,8 +288,21 @@ public List<BlockLocationInfo> getBlockLocations(URIStatus status)
         if (locations.isEmpty() && mFsContext.getPathConf(new AlluxioURI(status.getPath()))
             .getBoolean(PropertyKey.USER_UFS_BLOCK_LOCATION_ALL_FALLBACK_ENABLED)) {
           // Case 2: Fallback to add all workers to locations so some apps (Impala) won't panic.
```

Comment:
If this situation occurs, can you directly return the [default value defined here](https://github.com/apache/hadoop/blob/rel/release-3.3.0/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/fs/FileSystem.java#L834)? This is more friendly to Impala's file handle cache and data cache. This is because Impala will use consistent hash to schedule data scan fragment, and random block location returns will reduce the cache hit rate.

Reply:
Thanks for the review. I don't quite understand the change method you are talking about,  or is [this](https://github.com/Alluxio/alluxio/pull/18448/files#diff-f2f57582853d263397dabf21e474ec0c731dd50a445d81403c0a8932199176abL280) what you are talking about?